### PR TITLE
add test for CRI log path

### LINKED ISSUE #61945: Update `ContainerLogPath` test
The [ContainerLogPath test](https://github.com/kubernetes/kubernetes/blob/v1.11.0-alpha.0/test/e2e_node/log_path_test.go) should be updated to check CRI log paths.

### PR DESCRIPTION
**Pull request by fatsheep9146**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #61945

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note
NONE
```

### CODE REVIEW

File: test/e2e_node/log_path_test.go [hunk 1]
```diff
@@ -142,6 +144,51 @@ var _ = framework.KubeDescribe("ContainerLogPath [NodeConformance]", func() {
...
+					Spec: v1.PodSpec{
+						// this pod is expected to exit successfully
+						RestartPolicy: v1.RestartPolicyNever,
+						Containers: []v1.Container{
+							{
+								Image: busyboxImage,
+								Name:  checkCRIContName,
+								// If we find expected cri log file and contains right content, exit 0
+								// else, keep checking until test timeout
+								Command: []string{"sh", "-c", "while true; do if [ -e " + expectedCRILogFile + " ] && grep -q " + logString + " " + expectedCRILogFile + "; then exit 0; fi; sleep 1; done"},
```
Comment (feiskyer) [inline test/e2e_node/log_path_test.go]: maybe it's better to wait for pod success and check the log file afterward?

Comment (fatsheep9146) [reply test/e2e_node/log_path_test.go]: /cc @feiskyer

Comment (feiskyer) [reply test/e2e_node/log_path_test.go]: I see. Thanks for the explanation.

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Comment by fatsheep9146** [comment]
@feiskyer Test is done. What else should i do?

**Comment by feiskyer** [comment]
still need approval from @yujuhong

**Comment by fatsheep9146** [comment]
@Random-Liu Can you help to approve this？Thank you for your time

File: test/e2e_node/log_path_test.go [hunk 2]
```diff
@@ -69,6 +70,7 @@ var _ = framework.KubeDescribe("ContainerLogPath [NodeConformance]", func() {
 
 				logDirVolumeName := "log-dir-vol"
 				logDir := kubelet.ContainerLogsDir
+				logCRIDir := "/var/log/pods"
```
Comment (pontiyaraja) [inline test/e2e_node/log_path_test.go]: Create new test case, with helper function for both the existing and new CRI log path.

**Review by pontiyaraja — CHANGES_REQUESTED** [review]

**Comment by pontiyaraja** [comment]
@feiskyer can you add `/release-note-none` comment

File: test/e2e_node/log_path_test.go (hunk 2 above)
Comment (fatsheep9146) [reply test/e2e_node/log_path_test.go]: You mean just like the table-driven test, which is organized as following, 
```
tcs := []struct {
       in string
       exp string
} {
    {in:"in1", exp:"out1"},
    {in:"in2", exp:"out2"},
}

for i, tc := range tcs {
       out := testFunc(tc.in)
       if exp != out {
              t.Errorf(...)
       }
}

```

Comment (pontiyaraja) [reply test/e2e_node/log_path_test.go]: No, create individual test cases.

Comment (fatsheep9146) [reply test/e2e_node/log_path_test.go]: Sorry, I'm still confused with "individual test cases".

Does "individual test cases" mean like this ?

```
Context("printed log to stdout", func() {
    It("should print log to correct log path", func() {
         // run with common helper function
    }
    It("should print log to correct cri log path", func() {
         // run with common helper function
    }
}
```

Comment (fatsheep9146) [reply test/e2e_node/log_path_test.go]: ok，thank you，I will modify like this

**Comment by fatsheep9146** [comment]
@pontiyaraja @feiskyer 
Sorry for my delay，I have already adjusted the structure of the test code as @pontiyaraja said. 
Thank you for your review again.

**Review by pontiyaraja — APPROVED** [review]

**Comment by fatsheep9146** [comment]
@pontiyaraja 
Sorry, I want to rebase three commit to single one. After I have force-pushed, the lgtm label is automatically removed.
Can you approve it again?

File: test/e2e_node/log_path_test.go [hunk 3]
```diff
@@ -17,131 +17,156 @@ limitations under the License.
...
 				}
+			}
+
+			createAndWaitPod := func(pod *v1.Pod) error {
+				podClient.Create(pod)
+				return framework.WaitForPodSuccessInNamespace(f.ClientSet, pod.Name, f.Namespace.Name)
+			}
+
+			var logPodName string
+			BeforeEach(func() {
```
Comment (pontiyaraja) [inline test/e2e_node/log_path_test.go]: why two `BeforeEach`

**Review by pontiyaraja — CHANGES_REQUESTED** [review]

File: test/e2e_node/log_path_test.go (hunk 3 above)
Comment (fatsheep9146) [reply test/e2e_node/log_path_test.go]: I mainly learned from another e2e_test file.

https://github.com/kubernetes/kubernetes/blob/197fc67693c2391dcbc652fc185ba85b5ef82a8e/test/e2e_node/security_context_test.go#L40

https://github.com/kubernetes/kubernetes/blob/197fc67693c2391dcbc652fc185ba85b5ef82a8e/test/e2e_node/security_context_test.go#L226

I think, in the future, there may be other test case to be added to log_path_test, which may have some other different initializing operations than Context("printed log to stdout"). 

For example, if we want to test the file based logs, which may be considered in the future. 

But the first `BeforeEach`, which initialize variable podClient, is needed in every test case.

If my concern is not necessary , I can modify this into one `BeforeEach`

Comment (pontiyaraja) [reply test/e2e_node/log_path_test.go]: You are using single spec in this file, so you can use only one `BeforeEach`, this two `BeforeEach` will get execute in this spec, so you can make it one.

**Review by pontiyaraja — CHANGES_REQUESTED** [review]

File: test/e2e_node/log_path_test.go (hunk 3 above)
Comment (fatsheep9146) [reply test/e2e_node/log_path_test.go]: ok，I will change this

**Comment by fatsheep9146** [comment]
@pontiyaraja I have made it into one `BeforeEach`, all tests are passed.

**Review by pontiyaraja — APPROVED** [review]

**Comment by pontiyaraja** [comment]
@kubernetes/sig-storage-pr-reviews

**Pull request merged**